// @wallet/constants/src/currencies/index.ts
export const CURRENCY_SYMBOL = {
  USD: '$',
  JPY: '¥',
  AUD: '$',
  BRL: 'R$',
  CAD: '$',
  CHF: '₣',
  CLP: '$',
  CNY: '¥',
  CZK: 'Kč',
  DKK: 'kr',
  EUR: '€',
  GBP: '£',
  HKD: '$',
  HUF: 'Ft',
  IDR: 'Rp',
  ILS: '₪',
  INR: '₹',
  ISK: 'kr',
  KRW: '₩',
  MXN: '$',
  MYR: 'RM',
  NOK: 'kr',
  NZD: '$',
  PHP: '₱',
  PKR: '₨',
  PLN: 'zł',
  RUB: '₽',
  SEK: 'kr',
  SGD: '$',
  THB: '฿',
  TRY: '₺',
  TWD: 'NT$',
  ZAR: 'R',
  VND: 'đ'
}

export type FiatCurrencyKey = keyof typeof CURRENCY_SYMBOL

export const CURRENCY_DATA = [
  { name: 'VND', symbol: CURRENCY_SYMBOL.VND },
  { name: 'USD', symbol: CURRENCY_SYMBOL.USD },
  { name: 'CNY', symbol: CURRENCY_SYMBOL.CNY },
  { name: 'JPY', symbol: CURRENCY_SYMBOL.JPY },
  { name: 'THB', symbol: CURRENCY_SYMBOL.THB },
  { name: 'AUD', symbol: CURRENCY_SYMBOL.AUD },
  { name: 'BRL', symbol: CURRENCY_SYMBOL.BRL },
  { name: 'CAD', symbol: CURRENCY_SYMBOL.CAD },
  { name: 'CHF', symbol: CURRENCY_SYMBOL.CHF },
  { name: 'CLP', symbol: CURRENCY_SYMBOL.CLP },
  { name: 'CZK', symbol: CURRENCY_SYMBOL.CZK },
  { name: 'DKK', symbol: CURRENCY_SYMBOL.DKK },
  { name: 'EUR', symbol: CURRENCY_SYMBOL.EUR },
  { name: 'GBP', symbol: CURRENCY_SYMBOL.GBP },
  { name: 'HKD', symbol: CURRENCY_SYMBOL.HKD },
  { name: 'HUF', symbol: CURRENCY_SYMBOL.HUF },
  { name: 'IDR', symbol: CURRENCY_SYMBOL.IDR },
  { name: 'ILS', symbol: CURRENCY_SYMBOL.ILS },
  { name: 'INR', symbol: CURRENCY_SYMBOL.INR },
  { name: 'ISK', symbol: CURRENCY_SYMBOL.ISK },
  { name: 'KRW', symbol: CURRENCY_SYMBOL.KRW },
  { name: 'MXN', symbol: CURRENCY_SYMBOL.MXN },
  { name: 'MYR', symbol: CURRENCY_SYMBOL.MYR },
  { name: 'NOK', symbol: CURRENCY_SYMBOL.NOK },
  { name: 'NZD', symbol: CURRENCY_SYMBOL.NZD },
  { name: 'PHP', symbol: CURRENCY_SYMBOL.PHP },
  { name: 'PKR', symbol: CURRENCY_SYMBOL.PKR },
  { name: 'PLN', symbol: CURRENCY_SYMBOL.PLN },
  { name: 'RUB', symbol: CURRENCY_SYMBOL.RUB },
  { name: 'SEK', symbol: CURRENCY_SYMBOL.SEK },
  { name: 'SGD', symbol: CURRENCY_SYMBOL.SGD },
  { name: 'TRY', symbol: CURRENCY_SYMBOL.TRY },
  { name: 'TWD', symbol: CURRENCY_SYMBOL.TWD },
  { name: 'ZAR', symbol: CURRENCY_SYMBOL.ZAR }
]
