// @wallet/constants/src/chains/chainConfigCosmos.ts
const getBech32Prefix = (chain: string) => {
  return {
    bech32PrefixAccAddr: chain,
    bech32PrefixAccPub: chain + 'pub',
    bech32PrefixValAddr: chain + 'valoper',
    bech32PrefixValPub: chain + 'valoperpub',
    bech32PrefixConsAddr: chain + 'valcons',
    bech32PrefixConsPub: chain + 'valconspub'
  }
}

export const COSMOS_RPC_ENDPOINT = 'https://cosmos-rpc.rockrpc.net'
export const COSMOS_RPC_CONFIG = ''
export const COSMOS_REST_CONFIG = ''
// export const COSMOS_REST_ENDPOINT = 'https://lcd-cosmoshub.keplr.app'
export const COSMOS_REST_ENDPOINT = 'https://cosmos-lcd.quickapi.com:443'

export const OSMOSIS_RPC_ENDPOINT = 'https://rpc.osmosis.zone/'
export const OSMOSIS_RPC_CONFIG = ''
export const OSMOSIS_REST_ENDPOINT = 'https://lcd-osmosis.keplr.app'
export const OSMOSIS_REST_CONFIG = ''

export const SECRET_NETWORK_RPC_ENDPOINT = 'https://rpc-secret.keplr.app'
export const SECRET_NETWORK_RPC_CONFIG = ''
export const SECRET_NETWORK_REST_ENDPOINT = 'https://lcd-secret.keplr.app'
export const SECRET_NETWORK_REST_CONFIG = ''

export const AKASH_RPC_ENDPOINT = 'https://rpc-akash.keplr.app/'
export const AKASH_RPC_CONFIG = ''
export const AKASH_REST_ENDPOINT = 'https://lcd-akash.keplr.app'
export const AKASH_REST_CONFIG = ''

export const CRYPTO_ORG_RPC_ENDPOINT = ''
export const CRYPTO_ORG_RPC_CONFIG = ''
export const CRYPTO_ORG_REST_ENDPOINT = 'https://lcd-crypto-org.keplr.app'
export const CRYPTO_ORG_REST_CONFIG = ''

export const IOV_RPC_ENDPOINT = 'https://rpc-iov.keplr.app/'
export const IOV_RPC_CONFIG = ''
export const IOV_REST_ENDPOINT = ''
export const IOV_REST_CONFIG = ''

export const SIFCHAIN_RPC_ENDPOINT = 'https://rpc.sifchain.finance/'
export const SIFCHAIN_RPC_CONFIG = ''
export const SIFCHAIN_REST_ENDPOINT = 'https://lcd-sifchain.keplr.app'
export const SIFCHAIN_REST_CONFIG = ''

export const CERTIK_RPC_ENDPOINT = 'https://certik-mainnet-rpc.allthatnode.com'
export const CERTIK_RPC_CONFIG = ''
export const CERTIK_REST_ENDPOINT = 'https://lcd-certik.keplr.app'
export const CERTIK_REST_CONFIG = ''

export const IRIS_RPC_ENDPOINT = ''
export const IRIS_RPC_CONFIG = ''
export const IRIS_REST_ENDPOINT = 'https://lcd-iris.keplr.app'
export const IRIS_REST_CONFIG = ''

export const REGEN_RPC_ENDPOINT = ''
export const REGEN_RPC_CONFIG = ''
export const REGEN_REST_ENDPOINT = 'https://lcd-regen.keplr.app'
export const REGEN_REST_CONFIG = ''

export const PERSISTENCE_RPC_ENDPOINT = 'https://rpc.core.persistence.one/'
export const PERSISTENCE_RPC_CONFIG = ''
export const PERSISTENCE_REST_ENDPOINT = 'https://lcd-persistence.keplr.app'
export const PERSISTENCE_REST_CONFIG = ''

export const SENTINEL_RPC_ENDPOINT = 'https://rpc.sentinel.co/'
export const SENTINEL_RPC_CONFIG = ''
export const SENTINEL_REST_ENDPOINT = 'https://lcd-sentinel.keplr.app'
export const SENTINEL_REST_CONFIG = ''

export const KAVA_RPC_ENDPOINT = 'https://rpc.kava.io/'
export const KAVA_RPC_CONFIG = ''
export const KAVA_REST_ENDPOINT = 'https://api.data.kava.io'
export const KAVA_REST_CONFIG = ''

export const IXO_RPC_ENDPOINT = ''
export const IXO_RPC_CONFIG = ''
export const IXO_REST_ENDPOINT = 'https://lcd-impacthub.keplr.app'
export const IXO_REST_CONFIG = ''

export const EMONEY_RPC_ENDPOINT = 'https://rpc-emoney.keplr.app/'
export const EMONEY_RPC_CONFIG = ''
export const EMONEY_REST_ENDPOINT = 'https://lcd-emoney.keplr.app'
export const EMONEY_REST_CONFIG = ''

export const AGORIC_RPC_ENDPOINT = 'https://agoric-rpc.stakely.io'
export const AGORIC_RPC_CONFIG = ''
export const AGORIC_REST_ENDPOINT = 'https://lcd-agoric.keplr.app'
export const AGORIC_REST_CONFIG = ''

export const BETA_CYBER_NETWORK_RPC_ENDPOINT = ''
export const BETA_CYBER_NETWORK_RPC_CONFIG = ''
export const BETA_CYBER_NETWORK_REST_ENDPOINT = ''
export const BETA_CYBER_NETWORK_REST_CONFIG = ''

export const JUNO_RPC_ENDPOINT = 'https://rpc.juno.strange.love/'
export const JUNO_RPC_CONFIG = ''
export const JUNO_REST_ENDPOINT = 'https://lcd-juno.keplr.app'
export const JUNO_REST_CONFIG = ''

export const STARGAZE_RPC_ENDPOINT = 'https://rpc.stargaze-apis.com/'
export const STARGAZE_RPC_CONFIG = ''
export const STARGAZE_REST_ENDPOINT = 'https://lcd-stargaze.keplr.app'
export const STARGAZE_REST_CONFIG = ''

export const AXELAR_RPC_ENDPOINT = 'https://axelar-rpc.pops.one'
export const AXELAR_RPC_CONFIG = ''
export const AXELAR_REST_ENDPOINT = 'https://lcd-axelar.keplr.app'
export const AXELAR_REST_CONFIG = ''

export const SOMMELIER_RPC_ENDPOINT = ''
export const SOMMELIER_RPC_CONFIG = ''
export const SOMMELIER_REST_ENDPOINT = 'https://lcd-sommelier.keplr.app'
export const SOMMELIER_REST_CONFIG = ''

export const UMEE_RPC_ENDPOINT = 'https://umee-rpc.polkachu.com/'
export const UMEE_RPC_CONFIG = ''
export const UMEE_REST_ENDPOINT = 'https://umee-api.polkachu.com'
export const UMEE_REST_CONFIG = ''

// export const EVMOS_RPC_ENDPOINT = 'https://cosmos-rpc.rockrpc.net'
// export const EVMOS_RPC_CONFIG = ''
// export const EVMOS_REST_ENDPOINT = 'https://rest.bd.evmos.org:1317'
// export const EVMOS_REST_CONFIG = ''

export const EVMOS_RPC_ENDPOINT = 'https://evmos-rpc.publicnode.com:443'
export const EVMOS_RPC_CONFIG = ''
export const EVMOS_REST_ENDPOINT = 'https://lcd-evmos.keplr.app'
export const EVMOS_REST_CONFIG = ''

export const TERRA_RPC_ENDPOINT = 'https://terra-rpc.polkachu.com/'
export const TERRA_RPC_CONFIG = ''
export const TERRA_REST_ENDPOINT = 'https://lcd-phoenix.keplr.app'
export const TERRA_REST_CONFIG = ''

export const OLD_TERRA_RPC_ENDPOINT = 'https://terra-classic-rpc.publicnode.com/'
export const OLD_TERRA_RPC_CONFIG = ''
export const OLD_TERRA_REST_ENDPOINT = 'https://lcd-columbus.keplr.app'
export const OLD_TERRA_REST_CONFIG = ''

export const BAND_RPC_ENDPOINT = 'http://rpc.laozi1.bandchain.org/'
export const BAND_RPC_CONFIG = ''
export const BAND_REST_ENDPOINT = 'https://laozi1.bandchain.org/api'
export const BAND_REST_CONFIG = ''

export const COSMOS_DEFAULT_GAS: Record<string, number | string> = {
  send: 80000,
  intergrate: 250000
}

export const ChainConfigsCosmos = [
  {
    rpc: 'https://rpc.aura.network',
    rest: 'https://lcd.aura.network',
    chainId: 'aura_6322-2',
    chainName: 'Aura Xstaxy Mainnet',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('aura'),
    currencies: [
      {
        coinDenom: 'AURA',
        coinMinimalDenom: 'uaura',
        coinDecimals: 6,
        coinGeckoId: 'aura-network'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'AURA',
        coinMinimalDenom: 'uaura',
        coinDecimals: 6,
        coinGeckoId: 'aura-network'
      }
    ],
    stakeCurrency: {
      coinDenom: 'AURA',
      coinMinimalDenom: 'uaura',
      coinDecimals: 6,
      coinGeckoId: 'aura-network'
    },
    coinType: 118,
    gasPriceStep: {
      low: 0.0025,
      average: 0.0035,
      high: 0.0045
    },
    features: [
      'no-legacy-stdTx'
    ],
    walletUrlForStaking: 'https://aurascan.io/validators'
  },
  {
    rpc: COSMOS_RPC_ENDPOINT,
    rpcConfig: COSMOS_RPC_CONFIG,
    rest: COSMOS_REST_ENDPOINT,
    restConfig: COSMOS_REST_CONFIG,
    chainId: 'cosmoshub-4',
    chainName: 'Cosmos Hub',
    stakeCurrency: {
      coinDenom: 'ATOM',
      coinMinimalDenom: 'uatom',
      coinDecimals: 6,
      coinGeckoId: 'cosmos'
    },
    walletUrl: 'https://wallet.keplr.app/#/cosmoshub/stake',
    walletUrlForStaking: 'https://wallet.keplr.app/#/cosmoshub/stake',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('cosmos'),
    currencies: [
      {
        coinDenom: 'ATOM',
        coinMinimalDenom: 'uatom',
        coinDecimals: 6,
        coinGeckoId: 'cosmos'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'ATOM',
        coinMinimalDenom: 'uatom',
        coinDecimals: 6,
        coinGeckoId: 'cosmos'
      }
    ],
    coinType: 118,
    features: ['ibc-transfer', 'ibc-go'],
    gasPriceStep: {
      low: 0.00025,
      average: 0.0025,
      high: 0.025
    }
  },
  {
    rpc: OSMOSIS_RPC_ENDPOINT,
    rpcConfig: OSMOSIS_RPC_CONFIG,
    rest: OSMOSIS_REST_ENDPOINT,
    restConfig: OSMOSIS_REST_CONFIG,
    chainId: 'osmosis-1',
    chainName: 'Osmosis',
    stakeCurrency: {
      coinDenom: 'OSMO',
      coinMinimalDenom: 'uosmo',
      coinDecimals: 6,
      coinGeckoId: 'osmosis'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://app.osmosis.zone'
          : 'https://app.osmosis.zone',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/osmosis/stake'
          : 'http://localhost:8080/#/osmosis/stake',
    bip44: { coinType: 118 },
    bech32Config: getBech32Prefix('osmo'),
    currencies: [
      {
        coinDenom: 'OSMO',
        coinMinimalDenom: 'uosmo',
        coinDecimals: 6,
        coinGeckoId: 'osmosis'
      },
      {
        coinDenom: 'ION',
        coinMinimalDenom: 'uion',
        coinDecimals: 6,
        coinGeckoId: 'ion'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'OSMO',
        coinMinimalDenom: 'uosmo',
        coinDecimals: 6,
        coinGeckoId: 'osmosis'
      }
    ],
    gasPriceStep: {
      low: 0.0025,
      average: 0.025,
      high: 0.04
    },
    features: ['ibc-transfer', 'ibc-go', 'cosmwasm']
  },
  {
    rpc: SECRET_NETWORK_RPC_ENDPOINT,
    rpcConfig: SECRET_NETWORK_RPC_CONFIG,
    rest: SECRET_NETWORK_REST_ENDPOINT,
    restConfig: SECRET_NETWORK_REST_CONFIG,
    chainId: 'secret-4',
    chainName: 'Secret Network',
    stakeCurrency: {
      coinDenom: 'SCRT',
      coinMinimalDenom: 'uscrt',
      coinDecimals: 6,
      coinGeckoId: 'secret'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/secret/stake'
          : 'http://localhost:8080/#/secret/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/secret/stake'
          : 'http://localhost:8080/#/secret/stake',
    bip44: {
      coinType: 529
    },
    alternativeBIP44s: [
      {
        coinType: 118
      }
    ],
    bech32Config: getBech32Prefix('secret'),
    currencies: [
      {
        coinDenom: 'SCRT',
        coinMinimalDenom: 'uscrt',
        coinDecimals: 6,
        coinGeckoId: 'secret'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'SCRT',
        coinMinimalDenom: 'uscrt',
        coinDecimals: 6,
        coinGeckoId: 'secret'
      }
    ],
    coinType: 529,
    gasPriceStep: {
      low: 0.0125,
      average: 0.1,
      high: 0.25
    },
    features: ['secretwasm', 'ibc-go', 'ibc-transfer']
  },
  {
    rpc: AKASH_RPC_ENDPOINT,
    rpcConfig: AKASH_RPC_CONFIG,
    rest: AKASH_REST_ENDPOINT,
    restConfig: AKASH_REST_CONFIG,
    chainId: 'akashnet-2',
    chainName: 'Akash',
    stakeCurrency: {
      coinDenom: 'AKT',
      coinMinimalDenom: 'uakt',
      coinDecimals: 6,
      coinGeckoId: 'akash-network'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/akashnet/stake'
          : 'http://localhost:8080/#/akashnet/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/akashnet/stake'
          : 'http://localhost:8080/#/akashnet/stake',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('akash'),
    currencies: [
      {
        coinDenom: 'AKT',
        coinMinimalDenom: 'uakt',
        coinDecimals: 6,
        coinGeckoId: 'akash-network'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'AKT',
        coinMinimalDenom: 'uakt',
        coinDecimals: 6,
        coinGeckoId: 'akash-network'
      }
    ],
    features: ['ibc-transfer']
  },
  {
    rpc: CRYPTO_ORG_RPC_ENDPOINT,
    rpcConfig: CRYPTO_ORG_RPC_CONFIG,
    rest: CRYPTO_ORG_REST_ENDPOINT,
    restConfig: CRYPTO_ORG_REST_CONFIG,
    chainId: 'crypto-org-chain-mainnet-1',
    chainName: 'Crypto.org',
    stakeCurrency: {
      coinDenom: 'CRO',
      coinMinimalDenom: 'basecro',
      coinDecimals: 8,
      coinGeckoId: 'crypto-com-chain'
    },
    walletUrl: 'https://wallet.keplr.app/#/crypto-org/stake',
    walletUrlForStaking: 'https://wallet.keplr.app/#/crypto-org/stake',
    bip44: {
      coinType: 394
    },
    bech32Config: getBech32Prefix('cro'),
    currencies: [
      {
        coinDenom: 'CRO',
        coinMinimalDenom: 'basecro',
        coinDecimals: 8,
        coinGeckoId: 'crypto-com-chain'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'CRO',
        coinMinimalDenom: 'basecro',
        coinDecimals: 8,
        coinGeckoId: 'crypto-com-chain'
      }
    ],
    gasPriceStep: {
      low: 0.025,
      average: 0.03,
      high: 0.04
    },
    features: ['ibc-transfer']
  },
  {
    rpc: IOV_RPC_ENDPOINT,
    rpcConfig: IOV_RPC_CONFIG,
    rest: IOV_REST_ENDPOINT,
    restConfig: IOV_REST_CONFIG,
    chainId: 'iov-mainnet-ibc',
    chainName: 'Starname',
    stakeCurrency: {
      coinDenom: 'IOV',
      coinMinimalDenom: 'uiov',
      coinDecimals: 6,
      coinGeckoId: 'starname'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/iov-mainnet/stake'
          : 'http://localhost:8080/#/iov-mainnet/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/iov-mainnet/stake'
          : 'http://localhost:8080/#/iov-mainnet/stake',
    bip44: {
      coinType: 234
    },
    bech32Config: getBech32Prefix('star'),
    currencies: [
      {
        coinDenom: 'IOV',
        coinMinimalDenom: 'uiov',
        coinDecimals: 6,
        coinGeckoId: 'starname'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'IOV',
        coinMinimalDenom: 'uiov',
        coinDecimals: 6,
        coinGeckoId: 'starname'
      }
    ],
    gasPriceStep: {
      low: 1,
      average: 2,
      high: 3
    },
    features: ['ibc-transfer']
  },
  {
    rpc: SIFCHAIN_RPC_ENDPOINT,
    rpcConfig: SIFCHAIN_RPC_CONFIG,
    rest: SIFCHAIN_REST_ENDPOINT,
    restConfig: SIFCHAIN_REST_CONFIG,
    chainId: 'sifchain-1',
    chainName: 'Sifchain',
    stakeCurrency: {
      coinDenom: 'ROWAN',
      coinMinimalDenom: 'rowan',
      coinDecimals: 18,
      coinGeckoId: 'sifchain'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/sifchain/stake'
          : 'http://localhost:8080/#/sifchain/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/sifchain/stake'
          : 'http://localhost:8080/#/sifchain/stake',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('sif'),
    currencies: [
      {
        coinDenom: 'ROWAN',
        coinMinimalDenom: 'rowan',
        coinDecimals: 18,
        coinGeckoId: 'sifchain'
      },
      {
        coinDenom: 'Tether USDT',
        coinMinimalDenom: 'cusdt',
        coinDecimals: 6
      },
      {
        coinDenom: 'Ethereum',
        coinMinimalDenom: 'ceth',
        coinDecimals: 18
      },
      {
        coinDenom: 'Basic Attention Token',
        coinMinimalDenom: 'cbat',
        coinDecimals: 18
      },
      {
        coinDenom: 'Aragon',
        coinMinimalDenom: 'cant',
        coinDecimals: 18
      },
      {
        coinDenom: 'Bancor Network Token',
        coinMinimalDenom: 'cbnt',
        coinDecimals: 18
      },
      {
        coinDenom: '0x',
        coinMinimalDenom: 'czrx',
        coinDecimals: 18
      },
      {
        coinDenom: 'Chainlink',
        coinMinimalDenom: 'clink',
        coinDecimals: 18
      },
      {
        coinDenom: 'Decentraland',
        coinMinimalDenom: 'cmana',
        coinDecimals: 18
      },
      {
        coinDenom: 'Loopring',
        coinMinimalDenom: 'clrc',
        coinDecimals: 18
      },
      {
        coinDenom: 'Enjin Coin',
        coinMinimalDenom: 'cenj',
        coinDecimals: 18
      },
      {
        coinDenom: 'Synthetix Network Token',
        coinMinimalDenom: 'csnx',
        coinDecimals: 18
      },
      {
        coinDenom: 'TrueUSD',
        coinMinimalDenom: 'ctusd',
        coinDecimals: 18
      },
      {
        coinDenom: 'Ocean Protocol',
        coinMinimalDenom: 'cocean',
        coinDecimals: 18
      },
      {
        coinDenom: 'Fantom',
        coinMinimalDenom: 'cftm',
        coinDecimals: 18
      },
      {
        coinDenom: 'sUSD',
        coinMinimalDenom: 'csusd',
        coinDecimals: 18
      },
      {
        coinDenom: 'USD Coin',
        coinMinimalDenom: 'cusdc',
        coinDecimals: 6
      },
      {
        coinDenom: 'Crypto com Coin',
        coinMinimalDenom: 'ccro',
        coinDecimals: 8
      },
      {
        coinDenom: 'Wrapped Bitcoin',
        coinMinimalDenom: 'cwbtc',
        coinDecimals: 8
      },
      {
        coinDenom: 'Swipe',
        coinMinimalDenom: 'csxp',
        coinDecimals: 18
      },
      {
        coinDenom: 'Band Protocol',
        coinMinimalDenom: 'cband',
        coinDecimals: 18
      },
      {
        coinDenom: 'Dai Stablecoin',
        coinMinimalDenom: 'cdai',
        coinDecimals: 18
      },
      {
        coinDenom: 'Compound',
        coinMinimalDenom: 'ccomp',
        coinDecimals: 18
      },
      {
        coinDenom: 'UMA',
        coinMinimalDenom: 'cuma',
        coinDecimals: 18
      },
      {
        coinDenom: 'Balancer',
        coinMinimalDenom: 'cbal',
        coinDecimals: 18
      },
      {
        coinDenom: 'Yearn finance',
        coinMinimalDenom: 'cyfi',
        coinDecimals: 18
      },
      {
        coinDenom: 'Serum',
        coinMinimalDenom: 'csrm',
        coinDecimals: 6
      },
      {
        coinDenom: 'Cream',
        coinMinimalDenom: 'ccream',
        coinDecimals: 18
      },
      {
        coinDenom: 'SAND',
        coinMinimalDenom: 'csand',
        coinDecimals: 18
      },
      {
        coinDenom: 'Sushi',
        coinMinimalDenom: 'csushi',
        coinDecimals: 18
      },
      {
        coinDenom: 'Empty Set Dollar',
        coinMinimalDenom: 'cesd',
        coinDecimals: 18
      },
      {
        coinDenom: 'Uniswap',
        coinMinimalDenom: 'cuni',
        coinDecimals: 18
      },
      {
        coinDenom: 'Aave',
        coinMinimalDenom: 'caave',
        coinDecimals: 18
      },
      {
        coinDenom: 'BarnBridge',
        coinMinimalDenom: 'cbond',
        coinDecimals: 18
      },
      {
        coinDenom: 'Wrapped Filecoin',
        coinMinimalDenom: 'cwfil',
        coinDecimals: 18
      },
      {
        coinDenom: 'The Graph',
        coinMinimalDenom: 'cgrt',
        coinDecimals: 18
      },
      {
        coinDenom: 'Tokenlon',
        coinMinimalDenom: 'clon',
        coinDecimals: 18
      },
      {
        coinDenom: '1inch',
        coinMinimalDenom: 'c1inch',
        coinDecimals: 18
      },
      {
        coinDenom: 'THORChain ERC20',
        coinMinimalDenom: 'crune',
        coinDecimals: 18
      },
      {
        coinDenom: 'Secret ERC20',
        coinMinimalDenom: 'cwscrt',
        coinDecimals: 6
      },
      {
        coinDenom: 'IoTeX',
        coinMinimalDenom: 'ciotx',
        coinDecimals: 18
      },
      {
        coinDenom: 'Reef Finance',
        coinMinimalDenom: 'creef',
        coinDecimals: 18
      },
      {
        coinDenom: 'COCOS BCX',
        coinMinimalDenom: 'ccocos',
        coinDecimals: 18
      },
      {
        coinDenom: 'Keep Network',
        coinMinimalDenom: 'ckeep',
        coinDecimals: 18
      },
      {
        coinDenom: 'Origin Protocol',
        coinMinimalDenom: 'cogn',
        coinDecimals: 18
      },
      {
        coinDenom: 'ODAOfi',
        coinMinimalDenom: 'cdaofi',
        coinDecimals: 18
      },
      {
        coinDenom: 'Linear',
        coinMinimalDenom: 'clina',
        coinDecimals: 18
      },
      {
        coinDenom: '12Ships',
        coinMinimalDenom: 'ctshp',
        coinDecimals: 18
      },
      {
        coinDenom: 'B.20',
        coinMinimalDenom: 'cb20',
        coinDecimals: 18
      },
      {
        coinDenom: 'Akropolis',
        coinMinimalDenom: 'cakro',
        coinDecimals: 18
      },
      {
        coinDenom: 'Rio Fuel Token',
        coinMinimalDenom: 'crfuel',
        coinDecimals: 18
      },
      {
        coinDenom: 'Rally',
        coinMinimalDenom: 'crly',
        coinDecimals: 18
      },
      {
        coinDenom: 'Convergence',
        coinMinimalDenom: 'cconv',
        coinDecimals: 18
      },
      {
        coinDenom: 'Render Token',
        coinMinimalDenom: 'crndr',
        coinDecimals: 18
      },
      {
        coinDenom: 'PAID Network',
        coinMinimalDenom: 'cpaid',
        coinDecimals: 18
      },
      {
        coinDenom: 'Tidal',
        coinMinimalDenom: 'ctidal',
        coinDecimals: 18
      },
      {
        coinDenom: 'Axie Infinity',
        coinMinimalDenom: 'caxs',
        coinDecimals: 18
      },
      {
        coinDenom: 'BitSong',
        coinMinimalDenom: 'cbtsg',
        coinDecimals: 18
      },
      {
        coinDenom: 'Cosmostarter',
        coinMinimalDenom: 'ccsms',
        coinDecimals: 18
      },
      {
        coinDenom: 'Dfyn Network',
        coinMinimalDenom: 'cdfyn',
        coinDecimals: 18
      },
      {
        coinDenom: 'DinoSwap',
        coinMinimalDenom: 'cdino',
        coinDecimals: 18
      },
      {
        coinDenom: 'DinoX',
        coinMinimalDenom: 'cdnxc',
        coinDecimals: 18
      },
      {
        coinDenom: 'Don-key',
        coinMinimalDenom: 'cdon',
        coinDecimals: 18
      },
      {
        coinDenom: 'Ethernity Chain',
        coinMinimalDenom: 'cern',
        coinDecimals: 18
      },
      {
        coinDenom: 'Frax',
        coinMinimalDenom: 'cfrax',
        coinDecimals: 18
      },
      {
        coinDenom: 'Frax Share',
        coinMinimalDenom: 'cfxs',
        coinDecimals: 18
      },
      {
        coinDenom: 'Knit Finance',
        coinMinimalDenom: 'ckft',
        coinDecimals: 18
      },
      {
        coinDenom: 'Lido DAO',
        coinMinimalDenom: 'cldo',
        coinDecimals: 18
      },
      {
        coinDenom: 'Doge Killer',
        coinMinimalDenom: 'cleash',
        coinDecimals: 18
      },
      {
        coinDenom: 'LGCY Network',
        coinMinimalDenom: 'clgcy',
        coinDecimals: 18
      },
      {
        coinDenom: 'Polygon',
        coinMinimalDenom: 'cmatic',
        coinDecimals: 18
      },
      {
        coinDenom: 'Metis Token',
        coinMinimalDenom: 'cmetis',
        coinDecimals: 18
      },
      {
        coinDenom: 'Oh! Finance',
        coinMinimalDenom: 'coh',
        coinDecimals: 18
      },
      {
        coinDenom: 'Polkastarter',
        coinMinimalDenom: 'cpols',
        coinDecimals: 18
      },
      {
        coinDenom: 'Marlin',
        coinMinimalDenom: 'cpond',
        coinDecimals: 18
      },
      {
        coinDenom: 'Quickswap',
        coinMinimalDenom: 'cquick',
        coinDecimals: 18
      },
      {
        coinDenom: 'Railgun',
        coinMinimalDenom: 'crail',
        coinDecimals: 18
      },
      {
        coinDenom: 'StaFi rATOM',
        coinMinimalDenom: 'cratom',
        coinDecimals: 18
      },
      {
        coinDenom: 'Saito',
        coinMinimalDenom: 'csaito',
        coinDecimals: 18
      },
      {
        coinDenom: 'Shiba Inu',
        coinMinimalDenom: 'cshib',
        coinDecimals: 18
      },
      {
        coinDenom: 'Tokemak',
        coinMinimalDenom: 'ctoke',
        coinDecimals: 18
      },
      {
        coinDenom: 'UFO Gaming',
        coinMinimalDenom: 'cufo',
        coinDecimals: 18
      },
      {
        coinDenom: 'UST (ERC-20)',
        coinMinimalDenom: 'cust',
        coinDecimals: 18
      },
      {
        coinDenom: '0chain',
        coinMinimalDenom: 'czcn',
        coinDecimals: 18
      },
      {
        coinDenom: 'Unizen',
        coinMinimalDenom: 'czcx',
        coinDecimals: 18
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'ROWAN',
        coinMinimalDenom: 'rowan',
        coinDecimals: 18,
        coinGeckoId: 'sifchain'
      }
    ],
    gasPriceStep: {
      low: 1000000000000,
      average: 1500000000000,
      high: 2000000000000
    },
    features: []
  },
  {
    rpc: CERTIK_RPC_ENDPOINT,
    rpcConfig: CERTIK_RPC_CONFIG,
    rest: CERTIK_REST_ENDPOINT,
    restConfig: CERTIK_REST_CONFIG,
    chainId: 'shentu-2.2',
    chainName: 'Certik',
    stakeCurrency: {
      coinDenom: 'CTK',
      coinMinimalDenom: 'uctk',
      coinDecimals: 6,
      coinGeckoId: 'certik'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/shentu/stake'
          : 'http://localhost:8080/#/shentu/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/shentu/stake'
          : 'http://localhost:8080/#/shentu/stake',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('certik'),
    currencies: [
      {
        coinDenom: 'CTK',
        coinMinimalDenom: 'uctk',
        coinDecimals: 6,
        coinGeckoId: 'certik'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'CTK',
        coinMinimalDenom: 'uctk',
        coinDecimals: 6,
        coinGeckoId: 'certik'
      }
    ],
    features: ['ibc-transfer', 'ibc-go']
  },
  {
    rpc: IRIS_RPC_ENDPOINT,
    rpcConfig: IRIS_RPC_CONFIG,
    rest: IRIS_REST_ENDPOINT,
    restConfig: IRIS_REST_CONFIG,
    chainId: 'irishub-1',
    chainName: 'IRISnet',
    stakeCurrency: {
      coinDenom: 'IRIS',
      coinMinimalDenom: 'uiris',
      coinDecimals: 6,
      coinGeckoId: 'iris-network'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/irishub/stake'
          : 'http://localhost:8080/#/irishub/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/irishub/stake'
          : 'http://localhost:8080/#/irishub/stake',
    bip44: {
      coinType: 118
    },
    alternativeBIP44s: [
      {
        coinType: 566
      }
    ],
    bech32Config: getBech32Prefix('iaa'),
    currencies: [
      {
        coinDenom: 'IRIS',
        coinMinimalDenom: 'uiris',
        coinDecimals: 6,
        coinGeckoId: 'iris-network'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'IRIS',
        coinMinimalDenom: 'uiris',
        coinDecimals: 6,
        coinGeckoId: 'iris-network'
      }
    ],
    gasPriceStep: {
      low: 0.2,
      average: 0.3,
      high: 0.4
    },
    features: ['ibc-transfer', 'ibc-go']
  },
  {
    rpc: REGEN_RPC_ENDPOINT,
    rpcConfig: REGEN_RPC_CONFIG,
    rest: REGEN_REST_ENDPOINT,
    restConfig: REGEN_REST_CONFIG,
    chainId: 'regen-1',
    chainName: 'Regen',
    stakeCurrency: {
      coinDenom: 'REGEN',
      coinMinimalDenom: 'uregen',
      coinDecimals: 6,
      coinGeckoId: 'regen'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/regen/stake'
          : 'http://localhost:8080/#/regen/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/regen/stake'
          : 'http://localhost:8080/#/regen/stake',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('regen'),
    currencies: [
      {
        coinDenom: 'REGEN',
        coinMinimalDenom: 'uregen',
        coinDecimals: 6,
        coinGeckoId: 'regen'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'REGEN',
        coinMinimalDenom: 'uregen',
        coinDecimals: 6,
        coinGeckoId: 'regen'
      }
    ],
    features: ['ibc-go', 'ibc-transfer', 'ibc-go']
  },
  {
    rpc: PERSISTENCE_RPC_ENDPOINT,
    rpcConfig: PERSISTENCE_RPC_CONFIG,
    rest: PERSISTENCE_REST_ENDPOINT,
    restConfig: PERSISTENCE_REST_CONFIG,
    chainId: 'core-1',
    chainName: 'Persistence',
    stakeCurrency: {
      coinDenom: 'XPRT',
      coinMinimalDenom: 'uxprt',
      coinDecimals: 6,
      coinGeckoId: 'persistence'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/core/stake'
          : 'http://localhost:8080/#/core/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/core/stake'
          : 'http://localhost:8080/#/core/stake',
    bip44: {
      coinType: 750
    },
    bech32Config: getBech32Prefix('persistence'),
    currencies: [
      {
        coinDenom: 'XPRT',
        coinMinimalDenom: 'uxprt',
        coinDecimals: 6,
        coinGeckoId: 'persistence'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'XPRT',
        coinMinimalDenom: 'uxprt',
        coinDecimals: 6,
        coinGeckoId: 'persistence'
      }
    ],
    gasPriceStep: {
      low: 0,
      average: 0.025,
      high: 0.04
    },
    features: ['ibc-transfer', 'ibc-go']
  },
  {
    rpc: SENTINEL_RPC_ENDPOINT,
    rpcConfig: SENTINEL_RPC_CONFIG,
    rest: SENTINEL_REST_ENDPOINT,
    restConfig: SENTINEL_REST_CONFIG,
    chainId: 'sentinelhub-2',
    chainName: 'Sentinel',
    stakeCurrency: {
      coinDenom: 'DVPN',
      coinMinimalDenom: 'udvpn',
      coinDecimals: 6,
      coinGeckoId: 'sentinel'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/sentinel/stake'
          : 'http://localhost:8080/#/sentinel/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/sentinel/stake'
          : 'http://localhost:8080/#/sentinel/stake',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('sent'),
    currencies: [
      {
        coinDenom: 'DVPN',
        coinMinimalDenom: 'udvpn',
        coinDecimals: 6,
        coinGeckoId: 'sentinel'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'DVPN',
        coinMinimalDenom: 'udvpn',
        coinDecimals: 6,
        coinGeckoId: 'sentinel'
      }
    ],
    gasPriceStep: {
      low: 0.1,
      average: 0.25,
      high: 0.4
    },
    features: ['ibc-transfer']
  },
  {
    rpc: KAVA_RPC_ENDPOINT,
    rpcConfig: KAVA_RPC_CONFIG,
    rest: KAVA_REST_ENDPOINT,
    restConfig: KAVA_REST_CONFIG,
    chainId: 'kava_2222-10',
    chainName: 'Kava',
    stakeCurrency: {
      coinDenom: 'KAVA',
      coinMinimalDenom: 'ukava',
      coinDecimals: 6,
      coinGeckoId: 'kava'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/kava/stake'
          : 'http://localhost:8080/#/kava/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/kava/stake'
          : 'http://localhost:8080/#/kava/stake',
    bip44: { coinType: 459 },
    alternativeBIP44s: [{ coinType: 118 }],
    bech32Config: getBech32Prefix('kava'),
    currencies: [
      {
        coinDenom: 'KAVA',
        coinMinimalDenom: 'ukava',
        coinDecimals: 6,
        coinGeckoId: 'kava'
      },
      {
        coinDenom: 'SWP',
        coinMinimalDenom: 'swp',
        coinDecimals: 6,
        coinGeckoId: 'kava-swap'
      },
      {
        coinDenom: 'USDX',
        coinMinimalDenom: 'usdx',
        coinDecimals: 6,
        coinGeckoId: 'usdx'
      },
      {
        coinDenom: 'HARD',
        coinMinimalDenom: 'hard',
        coinDecimals: 6
      },
      {
        coinDenom: 'BNB',
        coinMinimalDenom: 'bnb',
        coinDecimals: 8
      },
      {
        coinDenom: 'BTCB',
        coinMinimalDenom: 'btcb',
        coinDecimals: 8
      },
      {
        coinDenom: 'BUSD',
        coinMinimalDenom: 'busd',
        coinDecimals: 8
      },
      {
        coinDenom: 'XRPB',
        coinMinimalDenom: 'xrpb',
        coinDecimals: 8
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'KAVA',
        coinMinimalDenom: 'ukava',
        coinDecimals: 6,
        coinGeckoId: 'kava'
      }
    ],
    gasPriceStep: {
      low: 0,
      average: 0.001,
      high: 0.25
    },
    coinType: 459
  },
  {
    rpc: IXO_RPC_ENDPOINT,
    rpcConfig: IXO_RPC_CONFIG,
    rest: IXO_REST_ENDPOINT,
    restConfig: IXO_REST_CONFIG,
    chainId: 'impacthub-3',
    chainName: 'ixo',
    stakeCurrency: {
      coinDenom: 'IXO',
      coinMinimalDenom: 'uixo',
      coinDecimals: 6
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/impacthub/stake'
          : 'http://localhost:8080/#/impacthub/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/impacthub/stake'
          : 'http://localhost:8080/#/impacthub/stake',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('ixo'),
    currencies: [
      {
        coinDenom: 'IXO',
        coinMinimalDenom: 'uixo',
        coinDecimals: 6
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'IXO',
        coinMinimalDenom: 'uixo',
        coinDecimals: 6
      }
    ],
    features: ['ibc-transfer']
  },
  {
    rpc: EMONEY_RPC_ENDPOINT,
    rpcConfig: EMONEY_RPC_CONFIG,
    rest: EMONEY_REST_ENDPOINT,
    restConfig: EMONEY_REST_CONFIG,
    chainId: 'emoney-3',
    chainName: 'e-Money',
    stakeCurrency: {
      coinDenom: 'NGM',
      coinMinimalDenom: 'ungm',
      coinDecimals: 6,
      coinGeckoId: 'e-money'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/emoney/stake'
          : 'http://localhost:8080/#/emoney/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/emoney/stake'
          : 'http://localhost:8080/#/emoney/stake',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('emoney'),
    currencies: [
      {
        coinDenom: 'NGM',
        coinMinimalDenom: 'ungm',
        coinDecimals: 6,
        coinGeckoId: 'e-money'
      },
      {
        coinDenom: 'EEUR',
        coinMinimalDenom: 'eeur',
        coinDecimals: 6,
        coinGeckoId: 'e-money-eur'
      },
      {
        coinDenom: 'EDKK',
        coinMinimalDenom: 'edkk',
        coinDecimals: 6
      },
      {
        coinDenom: 'ESEK',
        coinMinimalDenom: 'esek',
        coinDecimals: 6
      },
      {
        coinDenom: 'ENOK',
        coinMinimalDenom: 'enok',
        coinDecimals: 6
      },
      {
        coinDenom: 'ECHF',
        coinMinimalDenom: 'echf',
        coinDecimals: 6
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'NGM',
        coinMinimalDenom: 'ungm',
        coinDecimals: 6,
        coinGeckoId: 'e-money'
      }
    ],
    gasPriceStep: {
      low: 1,
      average: 1,
      high: 1
    },
    features: ['ibc-transfer']
  },
  {
    rpc: AGORIC_RPC_ENDPOINT,
    rpcConfig: AGORIC_RPC_CONFIG,
    rest: AGORIC_REST_ENDPOINT,
    restConfig: AGORIC_REST_CONFIG,
    chainId: 'agoric-3',
    chainName: 'Agoric',
    stakeCurrency: {
      coinDenom: 'BLD',
      coinMinimalDenom: 'ubld',
      coinDecimals: 6,
      coinGeckoId: 'agoric'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/agoric/stake'
          : 'http://localhost:8080/#/agoric/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/agoric/stake'
          : 'http://localhost:8080/#/agoric/stake',
    bip44: {
      // exactly coinType: 564
      coinType: 118
    },
    bech32Config: getBech32Prefix('agoric'),
    currencies: [
      {
        coinDenom: 'BLD',
        coinMinimalDenom: 'ubld',
        coinDecimals: 6,
        coinGeckoId: 'agoric'
      },
      {
        coinDenom: 'RUN',
        coinMinimalDenom: 'urun',
        coinDecimals: 6
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'RUN',
        coinMinimalDenom: 'urun',
        coinDecimals: 6
      }
    ],
    gasPriceStep: {
      low: 0.03,
      average: 0.05,
      high: 0.071
    },
    features: ['ibc-go']
  },
  {
    rpc: BETA_CYBER_NETWORK_RPC_ENDPOINT,
    rpcConfig: BETA_CYBER_NETWORK_RPC_CONFIG,
    rest: BETA_CYBER_NETWORK_REST_ENDPOINT,
    restConfig: BETA_CYBER_NETWORK_REST_CONFIG,
    chainId: 'bostrom',
    chainName: 'Bostrom',
    stakeCurrency: {
      coinDenom: 'BOOT',
      coinMinimalDenom: 'boot',
      coinDecimals: 0
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/bostrom/stake'
          : 'http://localhost:8080/#/bostrom/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/bostrom/stake'
          : 'http://localhost:8080/#/bostrom/stake',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('bostrom'),
    currencies: [
      {
        coinDenom: 'BOOT',
        coinMinimalDenom: 'boot',
        coinDecimals: 0
      },
      {
        coinDenom: 'H',
        coinMinimalDenom: 'hydrogen',
        coinDecimals: 0
      },
      {
        coinDenom: 'V',
        coinMinimalDenom: 'millivolt',
        coinDecimals: 3
      },
      {
        coinDenom: 'A',
        coinMinimalDenom: 'milliampere',
        coinDecimals: 3
      },
      {
        coinDenom: 'TOCYB',
        coinMinimalDenom: 'tocyb',
        coinDecimals: 0
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'BOOT',
        coinMinimalDenom: 'boot',
        coinDecimals: 0
      }
    ],
    gasPriceStep: {
      low: 0,
      average: 0.01,
      high: 0.02
    },
    features: ['ibc-transfer', 'cosmwasm', 'ibc-go']
  },
  {
    rpc: JUNO_RPC_ENDPOINT,
    rpcConfig: JUNO_RPC_CONFIG,
    rest: JUNO_REST_ENDPOINT,
    restConfig: JUNO_REST_CONFIG,
    chainId: 'juno-1',
    chainName: 'Juno',
    stakeCurrency: {
      coinDenom: 'JUNO',
      coinMinimalDenom: 'ujuno',
      coinDecimals: 6,
      coinGeckoId: 'juno-network'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/juno/stake'
          : 'http://localhost:8080/#/juno/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/juno/stake'
          : 'http://localhost:8080/#/juno/stake',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('juno'),
    currencies: [
      {
        coinDenom: 'JUNO',
        coinMinimalDenom: 'ujuno',
        coinDecimals: 6,
        coinGeckoId: 'juno-network'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'JUNO',
        coinMinimalDenom: 'ujuno',
        coinDecimals: 6,
        coinGeckoId: 'juno-network'
      }
    ],
    gasPriceStep: {
      low: 0.075,
      average: 0.125,
      high: 0.225
    },
    features: ['cosmwasm', 'ibc-transfer', 'ibc-go']
  },
  {
    rpc: STARGAZE_RPC_ENDPOINT,
    rpcConfig: STARGAZE_RPC_CONFIG,
    rest: STARGAZE_REST_ENDPOINT,
    restConfig: STARGAZE_REST_CONFIG,
    chainId: 'stargaze-1',
    chainName: 'Stargaze',
    stakeCurrency: {
      coinDenom: 'STARS',
      coinMinimalDenom: 'ustars',
      coinDecimals: 6,
      coinGeckoId: 'stargaze'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/stargaze/stake'
          : 'http://localhost:8080/#/stargaze/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/stargaze/stake'
          : 'http://localhost:8080/#/stargaze/stake',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('stars'),
    currencies: [
      {
        coinDenom: 'STARS',
        coinMinimalDenom: 'ustars',
        coinDecimals: 6,
        coinGeckoId: 'stargaze'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'STARS',
        coinMinimalDenom: 'ustars',
        coinDecimals: 6,
        coinGeckoId: 'stargaze'
      }
    ],
    features: ['ibc-transfer', 'ibc-go'],
    gasPriceStep: {
      low: 1,
      average: 1.1,
      high: 1.2
    }
  },
  {
    rpc: AXELAR_RPC_ENDPOINT,
    rpcConfig: AXELAR_RPC_CONFIG,
    rest: AXELAR_REST_ENDPOINT,
    restConfig: AXELAR_REST_CONFIG,
    chainId: 'axelar-dojo-1',
    chainName: 'Axelar',
    stakeCurrency: {
      coinDenom: 'AXL',
      coinMinimalDenom: 'uaxl',
      coinDecimals: 6
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/axelar/stake'
          : 'http://localhost:8080/#/axelar/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/axelar/stake'
          : 'http://localhost:8080/#/axelar/stake',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('axelar'),
    currencies: [
      {
        coinDenom: 'AXL',
        coinMinimalDenom: 'uaxl',
        coinDecimals: 6
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'AXL',
        coinMinimalDenom: 'uaxl',
        coinDecimals: 6
      }
    ],
    gasPriceStep: {
      low: 0.05,
      average: 0.075,
      high: 0.1
    },
    features: ['ibc-transfer', 'ibc-go']
  },
  {
    rpc: SOMMELIER_RPC_ENDPOINT,
    rpcConfig: SOMMELIER_RPC_CONFIG,
    rest: SOMMELIER_REST_ENDPOINT,
    restConfig: SOMMELIER_REST_CONFIG,
    chainId: 'sommelier-3',
    chainName: 'Sommelier',
    stakeCurrency: {
      coinDenom: 'SOMM',
      coinMinimalDenom: 'usomm',
      coinDecimals: 6,
      coinGeckoId: 'sommelier'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/sommelier/stake'
          : 'http://localhost:8080/#/sommelier/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/sommelier/stake'
          : 'http://localhost:8080/#/sommelier/stake',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('somm'),
    currencies: [
      {
        coinDenom: 'SOMM',
        coinMinimalDenom: 'usomm',
        coinDecimals: 6,
        coinGeckoId: 'sommelier'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'SOMM',
        coinMinimalDenom: 'usomm',
        coinDecimals: 6,
        coinGeckoId: 'sommelier'
      }
    ],
    features: ['ibc-transfer', 'ibc-go']
  },
  {
    rpc: UMEE_RPC_ENDPOINT,
    rpcConfig: UMEE_RPC_CONFIG,
    rest: UMEE_REST_ENDPOINT,
    restConfig: UMEE_REST_CONFIG,
    chainId: 'umee-1',
    chainName: 'Umee',
    stakeCurrency: {
      coinDenom: 'UMEE',
      coinMinimalDenom: 'uumee',
      coinDecimals: 6,
      coinGeckoId: 'umee'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/umee/stake'
          : 'http://localhost:8080/#/umee/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/umee/stake'
          : 'http://localhost:8080/#/umee/stake',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('umee'),
    currencies: [
      {
        coinDenom: 'UMEE',
        coinMinimalDenom: 'uumee',
        coinDecimals: 6,
        coinGeckoId: 'umee'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'UMEE',
        coinMinimalDenom: 'uumee',
        coinDecimals: 6
      }
    ],
    gasPriceStep: {
      low: 0.1,
      average: 0.5,
      high: 1
    },
    features: ['ibc-transfer', 'ibc-go']
  },
  {
    rpc: EVMOS_RPC_ENDPOINT,
    rpcConfig: EVMOS_RPC_CONFIG,
    rest: EVMOS_REST_ENDPOINT,
    restConfig: EVMOS_REST_CONFIG,
    chainId: 'evmos_9001-2',
    chainName: 'Evmos (Beta)',
    stakeCurrency: {
      coinDenom: 'EVMOS',
      coinMinimalDenom: 'aevmos',
      coinDecimals: 18
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/evmos/stake'
          : 'http://localhost:8080/#/evmos/stake',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/#/evmos/stake'
          : 'http://localhost:8080/#/evmos/stake',
    bip44: {
      coinType: 60
    },
    bech32Config: getBech32Prefix('evmos'),
    currencies: [
      {
        coinDenom: 'EVMOS',
        coinMinimalDenom: 'aevmos',
        coinDecimals: 18,
        coinGeckoId: 'evmos'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'EVMOS',
        coinMinimalDenom: 'aevmos',
        coinDecimals: 18
      }
    ],
    gasPriceStep: {
      low: 25000000000,
      average: 40000000000,
      high: 55000000000
    },
    features: ['ibc-transfer', 'ibc-go'],
    beta: true,
    defaultFee: 0.04,
    isEthereum: true
  },
  {
    rpc: 'https://fx-json.functionx.io/',
    rpcConfig: '',
    rest: 'https://fx-rest.functionx.io',
    restConfig: '',                                               
    chainId: 'fxcore',
    chainName: 'Function X',
    stakeCurrency: {
      coinDenom: 'FX',
      coinMinimalDenom: 'FX',
      coinDecimals: 18,
      coinGeckoId: 'fx-coin'
    },
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('fx'),
    currencies: [
      {
        coinDenom: 'FX',
        coinMinimalDenom: 'FX',
        coinDecimals: 18,
        coinGeckoId: 'fx-coin'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'FX',
        coinMinimalDenom: 'FX',
        coinDecimals: 18,
        coinGeckoId: 'fx-coin'
      }
    ],
    coinType: 118,
    features: ['ibc-transfer', 'ibc-go'],
    defaultFee: 0.8,
    gasPriceStep: {
      low: 4000000000000,
      average: 4000000000000,
      high: 4100000000000
    }
  },
  {
    rpc: 'https://rpc.wallet.pacific-1.sei.io',
    rpcConfig: '',
    rest: 'https://rest.wallet.pacific-1.sei.io',
    restConfig: '',
    chainId: 'pacific-1',
    chainName: 'Sei Mainnet',
    stakeCurrency: {
      coinDenom: 'SEI',
      coinMinimalDenom: 'usei',
      coinDecimals: 6,
      coinGeckoId: 'sei'
    },
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('sei'),
    currencies: [
      {
        coinDenom: 'SEI',
        coinMinimalDenom: 'usei',
        coinDecimals: 6,
        coinGeckoId: 'sei-network'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'SEI',
        coinMinimalDenom: 'usei',
        coinDecimals: 6,
        coinGeckoId: 'sei'
      }
    ],
    coinType: 118,
    features: ['ibc-transfer', 'ibc-go'],
    defaultFee: 0.01,
    gasPriceStep: {
      low: 0.1,
      average: 0.15,
      high: 0.2
    }
  },
  {
    rpc: 'https://rpc.atlantic-2.seinetwork.io/',
    rpcConfig: '',
    rest: 'https://rest.atlantic-2.seinetwork.io/',
    restConfig: '',
    chainId: 'atlantic-2',
    chainName: 'Sei Testnet',
    stakeCurrency: {
      coinDenom: 'SEI',
      coinMinimalDenom: 'usei',
      coinDecimals: 6,
      coinGeckoId: 'sei'
    },
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('sei'),
    currencies: [
      {
        coinDenom: 'SEI',
        coinMinimalDenom: 'usei',
        coinDecimals: 6
        // coinGeckoId: 'sei'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'SEI',
        coinMinimalDenom: 'usei',
        coinDecimals: 6
        // coinGeckoId: 'sei'
      }
    ],
    coinType: 118,
    features: ['ibc-transfer', 'ibc-go'],
    defaultFee: 0,
    gasPriceStep: {
      low: 1,
      average: 1.2,
      high: 1.5
    }
  },
  {
    rpc: 'https://stride-rpc.polkachu.com/',
    rest: 'https://stride-api.polkachu.com/',
    chainId: 'stride-1',
    chainName: 'Stride',
    stakeCurrency: {
      coinDenom: 'STRD',
      coinMinimalDenom: 'ustrd',
      coinDecimals: 6,
      coinGeckoId: 'stride'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/chains/stride'
          : 'http://localhost:8080/chains/stride',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/chains/stride'
          : 'http://localhost:8080/chains/stride',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('stride'),
    currencies: [
      {
        coinDenom: 'STRD',
        coinMinimalDenom: 'ustrd',
        coinDecimals: 6,
        coinGeckoId: 'stride'
      },
      {
        coinDenom: 'stATOM',
        coinMinimalDenom: 'stuatom',
        coinDecimals: 6
      },
      {
        coinDenom: 'stOSMO',
        coinMinimalDenom: 'stuosmo',
        coinDecimals: 6
      },
      {
        coinDenom: 'stJUNO',
        coinMinimalDenom: 'stujuno',
        coinDecimals: 6
      },
      {
        coinDenom: 'stSTARS',
        coinMinimalDenom: 'stustars',
        coinDecimals: 6
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'STRD',
        coinMinimalDenom: 'ustrd',
        coinDecimals: 6,
        coinGeckoId: 'stride',
        gasPriceStep: {
          low: 0,
          average: 0,
          high: 0.04
        }
      }
    ],
    features: ['ibc-transfer', 'ibc-go'],
    gasPriceStep: {
      low: 0.0005,
      average: 0.001,
      high: 0.002
    }
  },
  {
    rpc: 'https://rpc-stargaze.keplr.app',
    rest: 'https://lcd-stargaze.keplr.app',
    chainId: 'stargaze-1',
    chainName: 'Stargaze',
    stakeCurrency: {
      coinDenom: 'STARS',
      coinMinimalDenom: 'ustars',
      coinDecimals: 6,
      coinGeckoId: 'stargaze'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/chains/stargaze'
          : 'http://localhost:8080/chains/stargaze',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/chains/stargaze'
          : 'http://localhost:8080/chains/stargaze',
    bip44: {
      coinType: 118
    },
    bech32Config: getBech32Prefix('stars'),
    currencies: [
      {
        coinDenom: 'STARS',
        coinMinimalDenom: 'ustars',
        coinDecimals: 6,
        coinGeckoId: 'stargaze'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'STARS',
        coinMinimalDenom: 'ustars',
        coinDecimals: 6,
        coinGeckoId: 'stargaze'
      }
    ],
    features: ['ibc-transfer', 'ibc-go']
  },
  {
    bech32Config: {
      bech32PrefixAccAddr: 'kujira',
      bech32PrefixAccPub: 'kujirapub',
      bech32PrefixConsAddr: 'kujiravalcons',
      bech32PrefixConsPub: 'kujiravalconspub',
      bech32PrefixValAddr: 'kujiravaloper',
      bech32PrefixValPub: 'kujiravaloperpub'
    },
    beta: true,
    bip44: {
      coinType: 118
    },
    chainId: 'kaiyo-1',
    chainName: 'Kujira',
    coinType: 118,
    currencies: [
      {
        coinDecimals: 6,
        coinDenom: 'KUJI',
        coinGeckoId: 'kujira',
        coinMinimalDenom: 'ukuji'
      },
      {
        coinDecimals: 6,
        coinDenom: 'USK',
        coinGeckoId: 'usk',
        coinMinimalDenom: 'factory/kujira1qk00h5atutpsv900x202pxx42npjr9thg58dnqpa72f2p7m2luase444a7/uusk'
      },
      {
        coinDecimals: 6,
        coinDenom: 'axlUSDC',
        coinGeckoId: 'usdc',
        coinMinimalDenom: 'ibc/295548A78785A1007F232DE286149A6FF512F180AF5657780FC89C009E2C348F'
      },
      {
        coinDecimals: 6,
        coinDenom: 'ATOM',
        coinGeckoId: 'cosmoshub',
        coinMinimalDenom: 'ibc/27394FB092D2ECCD56123C74F36E4C1F926001CEADA9CA97EA622B25F41E5EB2'
      },
      {
        coinDecimals: 6,
        coinDenom: 'CMDX',
        coinGeckoId: 'comdex',
        coinMinimalDenom: 'ibc/3607EB5B5E64DD1C0E12E07F077FF470D5BC4706AFCBC98FE1BA960E5AE4CE07'
      },
      {
        coinDecimals: 18,
        coinDenom: 'EVMOS',
        coinGeckoId: 'evmos',
        coinMinimalDenom: 'ibc/F3AA7EF362EC5E791FE78A0F4CCC69FEE1F9A7485EB1A8CAB3F6601C00522F10'
      },
      {
        coinDecimals: 6,
        coinDenom: 'JUNO',
        coinGeckoId: 'juno',
        coinMinimalDenom: 'ibc/EFF323CC632EC4F747C61BCE238A758EFDB7699C3226565F7C20DA06509D59A5'
      },
      {
        coinDecimals: 6,
        coinDenom: 'OSMO',
        coinGeckoId: 'osmosis',
        coinMinimalDenom: 'ibc/47BD209179859CDE4A2806763D7189B6E6FE13A17880FE2B42DE1E6C1E329E23'
      },
      {
        coinDecimals: 6,
        coinDenom: 'SCRT',
        coinGeckoId: 'secret',
        coinMinimalDenom: 'ibc/A358D7F19237777AF6D8AD0E0F53268F8B18AE8A53ED318095C14D6D7F3B2DB5'
      },
      {
        coinDecimals: 6,
        coinDenom: 'STARS',
        coinGeckoId: 'stars',
        coinMinimalDenom: 'ibc/4F393C3FCA4190C0A6756CE7F6D897D5D1BE57D6CCB80D0BC87393566A7B6602'
      },
      {
        coinDecimals: 18,
        coinDenom: 'wAVAX',
        coinGeckoId: 'avalanche-2',
        coinMinimalDenom: 'ibc/004EBF085BBED1029326D56BE8A2E67C08CECE670A94AC1947DF413EF5130EB2'
      },
      {
        coinDecimals: 18,
        coinDenom: 'wETH',
        coinGeckoId: 'ethereum',
        coinMinimalDenom: 'ibc/1B38805B1C75352B28169284F96DF56BDEBD9E8FAC005BDCC8CF0378C82AA8E7'
      }
    ],
    feeCurrencies: [
      {
        coinDecimals: 6,
        coinDenom: 'KUJI',
        coinGeckoId: 'kujira',
        coinMinimalDenom: 'ukuji'
      }
    ],
    // rest: 'https://rest.cosmos.directory/kujira/',
    // rpc: 'https://kujira-rpc.nodes.defiantlabs.net',
    rest: 'https://kujira-rest.publicnode.com/',
    rpc: 'https://kujira-rpc.publicnode.com:443',
    stakeCurrency: {
      coinDecimals: 6,
      coinDenom: 'KUJI',
      coinGeckoId: 'kujira',
      coinMinimalDenom: 'ukuji'
    },
    gasPriceStep: {
      low: 0.01,
      average: 0.025,
      high: 0.04
    }
  },
  {
    bech32Config: {
      bech32PrefixAccAddr: 'inj',
      bech32PrefixAccPub: 'injpub',
      bech32PrefixConsAddr: 'injvalcons',
      bech32PrefixConsPub: 'injvalconspub',
      bech32PrefixValAddr: 'injvaloper',
      bech32PrefixValPub: 'injvaloperpub'
    },
    beta: true,
    bip44: {
      // coinType: 60
      // Coin98 Select for same cosmos path
      coinType: 118
    },
    chainId: 'injective-1',
    chainName: 'Injective v1.1',
    currencies: [
      {
        coinDecimals: 18,
        coinDenom: 'INJ',
        coinGeckoId: 'injective-protocol',
        coinMinimalDenom: 'inj'
      }
    ],
    features: [
      'ibc-transfer',
      'ibc-go',
      'eth-address-gen',
      'eth-key-sign'
    ],
    feeCurrencies: [
      {
        coinDecimals: 18,
        coinDenom: 'INJ',
        coinGeckoId: 'injective-protocol',
        coinMinimalDenom: 'inj',
        gasPriceStep: {
          low: 500000000,
          average: 2500000000,
          high: 5000000000
        }
      }
    ],
    rest: 'https://injective-rest.publicnode.com',
    rpc: 'https://injective-rpc.publicnode.com:443',
    stakeCurrency: {
      coinDecimals: 18,
      coinDenom: 'INJ',
      coinGeckoId: 'injective-protocol',
      coinMinimalDenom: 'inj'
    },
    walletUrl: 'https://hub.injective.network/',
    walletUrlForStaking: 'https://hub.injective.network/',
    // Support Ethermint
    isEthereum: true,
    gasPriceStep: {
      low: 500000000,
      average: 2500000000,
      high: 5000000000
    }
  },
  {
    defaultFee: 0.25,
    chainId: 'archway-1',
    chainName: 'Constantine Mainnet',
    rpc: 'https://rpc.mainnet.archway.io',
    rest: 'https://api.mainnet.archway.io',
    stakeCurrency: {
      coinDenom: 'ARCH',
      coinMinimalDenom: 'aarch',
      coinDecimals: 18
    },
    bip44: {
      coinType: 118
    },
    bech32Config: {
      bech32PrefixAccAddr: 'archway',
      bech32PrefixAccPub: 'archwaypub',
      bech32PrefixValAddr: 'archwayvaloper',
      bech32PrefixValPub: 'archwayvaloperpub',
      bech32PrefixConsAddr: 'archwayvalcons',
      bech32PrefixConsPub: 'archwayvalconspub'
    },
    currencies: [{
      coinDenom: 'ARCH',
      coinMinimalDenom: 'aarch',
      coinDecimals: 18,
      coinGeckoId: 'archway'
    }],
    feeCurrencies: [{
      coinDenom: 'ARCH',
      coinMinimalDenom: 'aarch',
      coinDecimals: 18
    }],
    coinType: 118,
    gasPriceStep: {
      low: 1000000000000,
      average: 1500000000000,
      high: 2000000000000
    },
    faucets: ['https://faucet.constantine.archway.tech'],
    features: ['cosmwasm']
  },
  {
    defaultFee: 0.25,
    chainId: 'constantine-3',
    chainName: 'Constantine Testnet',
    rpc: 'https://rpc.constantine.archway.tech:443',
    rest: 'https://api.constantine.archway.tech',
    stakeCurrency: {
      coinDenom: 'CONST',
      coinMinimalDenom: 'aconst',
      coinDecimals: 18
    },
    bip44: {
      coinType: 118
    },
    bech32Config: {
      bech32PrefixAccAddr: 'archway',
      bech32PrefixAccPub: 'archwaypub',
      bech32PrefixValAddr: 'archwayvaloper',
      bech32PrefixValPub: 'archwayvaloperpub',
      bech32PrefixConsAddr: 'archwayvalcons',
      bech32PrefixConsPub: 'archwayvalconspub'
    },
    currencies: [{
      coinDenom: 'CONST',
      coinMinimalDenom: 'aconst',
      coinDecimals: 18
    }],
    feeCurrencies: [{
      coinDenom: 'CONST',
      coinMinimalDenom: 'aconst',
      coinDecimals: 18
    }],
    coinType: 118,
    gasPriceStep: {
      low: 1000000000000,
      average: 1500000000000,
      high: 2000000000000
    },
    faucets: ['https://faucet.constantine.archway.tech'],
    features: ['cosmwasm']
  },
  {
    rpc: 'https://rpc-agoric.keplr.app',
    rest: 'https://lcd-agoric.keplr.app',
    chainId: 'agoric-3',
    chainName: 'Agoric',
    stakeCurrency: {
      coinDenom: 'BLD',
      coinMinimalDenom: 'ubld',
      coinDecimals: 6,
      coinGeckoId: 'agoric'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/chains/agoric'
          : 'http://localhost:8080/chains/agoric',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/chains/agoric'
          : 'http://localhost:8080/chains/agoric',
    bip44: {
      coinType: 564
    },
    bech32Config: getBech32Prefix('agoric'),
    currencies: [
      {
        coinDenom: 'BLD',
        coinMinimalDenom: 'ubld',
        coinDecimals: 6,
        coinGeckoId: 'agoric'
      },
      {
        coinDenom: 'IST',
        coinMinimalDenom: 'uist',
        coinDecimals: 6
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'IST',
        coinMinimalDenom: 'uist',
        coinDecimals: 6,
        gasPriceStep: {
          low: 0,
          average: 0,
          high: 0
        }
      }
    ],
    features: ['ibc-go']
  },
  {
    rpc: TERRA_RPC_ENDPOINT,
    rest: TERRA_REST_ENDPOINT,
    chainId: 'phoenix-1',
    chainName: 'terra2',
    stakeCurrency: {
      coinDenom: 'LUNA',
      coinMinimalDenom: 'uluna',
      coinDecimals: 6,
      coinGeckoId: 'terra-luna-2'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/chains/terra'
          : 'http://localhost:8080/chains/terra',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/chains/terra'
          : 'http://localhost:8080/chains/terra',
    bip44: { coinType: 330 },
    bech32Config: getBech32Prefix('terra'),
    currencies: [
      {
        coinDenom: 'LUNA',
        coinMinimalDenom: 'uluna',
        coinDecimals: 6,
        coinGeckoId: 'terra-luna-2'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'LUNA',
        coinMinimalDenom: 'uluna',
        coinDecimals: 6,
        coinGeckoId: 'terra-luna-2'
      }
    ],
    gasPriceStep: {
      low: 0.015,
      average: 0.025,
      high: 0.04
    },
    features: ['ibc-transfer', 'ibc-go', 'cosmwasm']
  },
  {
    rpc: OLD_TERRA_RPC_ENDPOINT,
    rest: OLD_TERRA_REST_ENDPOINT,
    chainId: 'columbus-5',
    chainName: 'terra',
    stakeCurrency: {
      coinDenom: 'LUNA',
      coinMinimalDenom: 'uluna',
      coinDecimals: 6,
      coinGeckoId: 'terra-luna'
    },
    walletUrl:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/chains/terra'
          : 'http://localhost:8080/chains/terra',
    walletUrlForStaking:
        process.env.NODE_ENV === 'production'
          ? 'https://wallet.keplr.app/chains/terra'
          : 'http://localhost:8080/chains/terra',
    bip44: { coinType: 330 },
    bech32Config: getBech32Prefix('terra'),
    currencies: [
      {
        coinDenom: 'LUNC',
        coinMinimalDenom: 'uluna',
        coinDecimals: 6,
        coinGeckoId: 'terra-luna'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'LUNC',
        coinMinimalDenom: 'uluna',
        coinDecimals: 6,
        coinGeckoId: 'terra-luna'
      }
    ],
    gasPriceStep: {
      low: 32,
      average: 32,
      high: 32
    },
    features: ['ibc-transfer', 'ibc-go', 'cosmwasm']
  },
  {
    chainId: 'xion-testnet-1',
    chainName: 'xion-testnet-1',
    rpc: 'https://rpc.xion-testnet-1.burnt.com:443',
    rest: 'https://api.xion-testnet-1.burnt.com',
    bip44: {
      coinType: 118
    },
    coinType: 118,
    bech32Config: {
      bech32PrefixAccAddr: 'xion',
      bech32PrefixAccPub: 'xionpub',
      bech32PrefixValAddr: 'xionvaloper',
      bech32PrefixValPub: 'xionvaloperpub',
      bech32PrefixConsAddr: 'xionvalcons',
      bech32PrefixConsPub: 'xionvalconspub'
    },
    currencies: [
      {
        coinDenom: 'XION',
        coinMinimalDenom: 'uxion',
        coinDecimals: 6,
        coinGeckoId: 'xion'
      }
    ],
    feeCurrencies: [
      {
        coinDenom: 'XION',
        coinMinimalDenom: 'uxion',
        coinDecimals: 6,
        coinGeckoId: 'xion',
        gasPriceStep: {
          low: 0.01,
          average: 0.025,
          high: 0.03
        }
      }
    ],
    gasPriceStep: {
      low: 0.01,
      average: 0.025,
      high: 0.03
    },
    stakeCurrency: {
      coinDenom: 'XION',
      coinMinimalDenom: 'uxion',
      coinDecimals: 6,
      coinGeckoId: 'xion'
    },
    features: [
      'ibc-go',
      'ibc-transfer',
      'no-legacy-stdTx'
    ]
  },
  {
    bech32Config: {
      bech32PrefixAccAddr: 'celestia',
      bech32PrefixAccPub: 'celestiapub',
      bech32PrefixConsAddr: 'celestiavalcons',
      bech32PrefixConsPub: 'celestiavalconspub',
      bech32PrefixValAddr: 'celestiavaloper',
      bech32PrefixValPub: 'celestiavaloperpub'
    },
    bip44: {
      coinType: 118
    },
    chainId: 'celestia',
    chainName: 'Celestia',
    currencies: [
      {
        coinDecimals: 6,
        coinDenom: 'TIA',
        coinGeckoId: 'celestia',
        coinMinimalDenom: 'utia'
      }
    ],
    features: [],
    feeCurrencies: [
      {
        coinDecimals: 6,
        coinDenom: 'TIA',
        coinGeckoId: 'celestia',
        coinMinimalDenom: 'utia',
        gasPriceStep: {
          low: 0.01,
          average: 0.02,
          high: 0.1
        }
      }
    ],
    rpc: 'https://celestia-rpc.polkachu.com',
    rest: 'https://public-celestia-lcd.numia.xyz',
    stakeCurrency: {
      coinDecimals: 6,
      coinDenom: 'TIA',
      coinGeckoId: 'celestia',
      coinMinimalDenom: 'utia'
    },
    gasPriceStep: {
      low: 0.01,
      average: 0.02,
      high: 0.1
    }
  }
  // {
  //   rpc: BAND_RPC_ENDPOINT,
  //   rest: BAND_REST_ENDPOINT,
  //   chainId: 'laozi-mainnet',
  //   chainName: 'band',
  //   stakeCurrency: {
  //     coinDenom: 'band',
  //     coinMinimalDenom: 'uband',
  //     coinDecimals: 6,
  //     coinGeckoId: 'band-protocol'
  //   },
  //   walletUrl:
  //       process.env.NODE_ENV === 'production'
  //         ? 'https://wallet.keplr.app/chains/band'
  //         : 'http://localhost:8080/chains/band',
  //   walletUrlForStaking:
  //       process.env.NODE_ENV === 'production'
  //         ? 'https://wallet.keplr.app/chains/band'
  //         : 'http://localhost:8080/chains/band',
  //   bip44: { coinType: 494 },
  //   bech32Config: getBech32Prefix('band'),
  //   currencies: [
  //     {
  //       coinDenom: 'band',
  //       coinMinimalDenom: 'uband',
  //       coinDecimals: 6,
  //       coinGeckoId: 'band-protocol'
  //     }
  //   ],
  //   feeCurrencies: [
  //     {
  //       coinDenom: 'band',
  //       coinMinimalDenom: 'uband',
  //       coinDecimals: 6,
  //       coinGeckoId: 'band-protocol'
  //     }
  //   ],
  //   gasPriceStep: {
  //     low: 0.0025,
  //     average: 0.025,
  //     high: 0.25
  //   },
  //   features: ['ibc-transfer', 'ibc-go', 'cosmwasm']
  // }

]
