// @wallet/constants/src/exchange/index.ts
import type { ExchangeType, FixedStableList, FixedTokenList, IInfoExchange, SuperLinkAddress } from '@wallet/core'
import { CHAIN_DATA, CHAIN_TYPE } from '../chains'
import { getChainImage } from '@wallet/utils/src/image'
import { get } from 'lodash'

export const genSwapInfoChain = (chain: string): IInfoExchange => {
  const chainData = CHAIN_DATA[chain]

  return {
    image: chainData.icon as string,
    name: chainData.name,
    shortName: chainData.shortName as string
  }
}

export const SUPERLINK_ADDRESS: SuperLinkAddress = {
  [CHAIN_TYPE.ether]: '0xD62707f513F4B472d0BaDd922D74c49e06d9e7c7',
  [CHAIN_TYPE.binanceSmart]: '0xB1b9242DCB1A76F5682dA4ef98DB44897d4b2f39',
  [CHAIN_TYPE.avax]: '0xf9D2719295D7AF60614047399C3b72c1F2a1EE1b',
  [CHAIN_TYPE.matic]: '0xA89c387783Ec939e81004e87C04A67eDF2321EbD',
  // [CHAIN_TYPE.celo]: '0x836bf46520C373Fdc4dc7E5A3bAe735d13bD44e3',
  // [CHAIN_TYPE.kucoin]: '0xb0E042db37B5d9D90A059B749cca33Ad9adB7237',
  [CHAIN_TYPE.fantom]: '0xdD20ee2ADD6682585EBAAe2690ab2e328Ee501fC',
  // [CHAIN_TYPE.boba]: '0xc4cAd0938256ABA4417c565044Be2c2EB77096cb',
  [CHAIN_TYPE.heco]: '0x0e1D41001B5847c6CE6cAcbF75b3A5eA3bB0f034',
  // [CHAIN_TYPE.harmony]: '0x82C3dA62b7dB06E93c67bA90330cCEDEef725A63',
  [CHAIN_TYPE.arbitrum]: '0x2E1D30460265bFEBedacf5bb6f9A80F0E74B7498',
  [CHAIN_TYPE.solana]: '',
  [CHAIN_TYPE.ton]: '',
  [CHAIN_TYPE.optimism]: '0x5226f9Abc29CbFfE94D7F6bac06B4C10d9000Acf',
  [CHAIN_TYPE.tomo]: '',
  [CHAIN_TYPE.base]: '',
  allChain:''

}

export const EXCHANGE_TYPE: ExchangeType = {
  spaceGate: {
    type: 'spaceGate',
    isNoConfig: true,
    info: {
      image: 'app_spacegate',
      name: 'SpaceGate',
      shortName: 'Multichain'
    }
  },
  [CHAIN_TYPE.ether]: {
    type: 'defi',
    address: SUPERLINK_ADDRESS[CHAIN_TYPE.ether],
    info: genSwapInfoChain(CHAIN_TYPE.ether)
  },
  [CHAIN_TYPE.binanceSmart]: {
    type: 'defi',
    address: SUPERLINK_ADDRESS[CHAIN_TYPE.binanceSmart],
    info: genSwapInfoChain(CHAIN_TYPE.binanceSmart)
  },
  [CHAIN_TYPE.avax]: {
    type: 'defi',
    address: SUPERLINK_ADDRESS[CHAIN_TYPE.avax],
    info: genSwapInfoChain(CHAIN_TYPE.avax)
  },
  [CHAIN_TYPE.matic]: {
    type: 'defi',
    address: SUPERLINK_ADDRESS[CHAIN_TYPE.matic],
    info: genSwapInfoChain(CHAIN_TYPE.matic)
  },
  // [CHAIN_TYPE.fantom]: {
  //   type: 'defi',
  //   address: SUPERLINK_ADDRESS[CHAIN_TYPE.fantom],
  //   info: genSwapInfoChain(CHAIN_TYPE.fantom)
  // },
  // [CHAIN_TYPE.heco]: {
  //   type: 'defi',
  //   address: SUPERLINK_ADDRESS[CHAIN_TYPE.heco],
  //   info: genSwapInfoChain(CHAIN_TYPE.heco)
  // }
  [CHAIN_TYPE.arbitrum]: {
    type: 'defi',
    address: SUPERLINK_ADDRESS[CHAIN_TYPE.arbitrum],
    info: genSwapInfoChain(CHAIN_TYPE.arbitrum)
  },
  // chua có token
  [CHAIN_TYPE.optimism]: {
    type: 'defi',
    address: SUPERLINK_ADDRESS[CHAIN_TYPE.optimism],
    info: genSwapInfoChain(CHAIN_TYPE.optimism)
  },
  [CHAIN_TYPE.solana]: {
    type: 'defi',
    address: SUPERLINK_ADDRESS[CHAIN_TYPE.solana],
    info: genSwapInfoChain(CHAIN_TYPE.solana)
  },
  // [CHAIN_TYPE.ton]: {
  //   type: 'defi',
  //   address: SUPERLINK_ADDRESS[CHAIN_TYPE.ton],
  //   info: genSwapInfoChain(CHAIN_TYPE.ton)
  // },
  [CHAIN_TYPE.tomo]: {
    type: 'defi',
    address: SUPERLINK_ADDRESS[CHAIN_TYPE.tomo],
    info: genSwapInfoChain(CHAIN_TYPE.tomo)
  },
  [CHAIN_TYPE.base]: {
    type: 'defi',
    address: SUPERLINK_ADDRESS[CHAIN_TYPE.base],
    info: genSwapInfoChain(CHAIN_TYPE.base)
  },
}

export const FIXED_TOKENS_BY_CHAIN: FixedTokenList = {
  [CHAIN_TYPE.ether]: {
    chain: CHAIN_TYPE.ether,
    from: {
      cgkId: 'ethereum',
      chainId: 1,
      chain: CHAIN_TYPE.ether,
      name: 'Ether',
      symbol: 'ETH',
      decimals: 18,
      image: getChainImage(CHAIN_TYPE.ether),
      wrapped: '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2'
    },
    to: {
      chainId: 1,
      chain: CHAIN_TYPE.ether,
      cgkId: 'coin98',
      symbol: 'C98',
      decimals: 18,
      name: 'Coin98',
      image: 'https://coin98.s3.amazonaws.com/uwqLJuMfd8fsehgM',
      address: '0xAE12C5930881c53715B369ceC7606B70d8EB229f'
    },
    pairs: [
      { address: '', symbol: 'ETH' },
      { address: '0x6B175474E89094C44Da98b954EedeAC495271d0F', symbol: 'DAI' },
      { address: '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48', symbol: 'USDC' },
      { address: '0xdAC17F958D2ee523a2206206994597C13D831ec7', symbol: 'USDT' }
    ]
  },

  allChain: {
    chain: CHAIN_TYPE.binanceSmart,
    from: {
      cgkId: 'binancecoin',
      chainId: 56,
      name: 'BNB Token',
      symbol: 'BNB',
      decimals: 18,
      image: getChainImage(CHAIN_TYPE.binanceSmart),
      wrapped: '0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c'
    },
    to: {
      chainId: 56,
      chain: CHAIN_TYPE.binanceSmart,
      cgkId: 'coin98',
      symbol: 'C98',
      decimals: 18,
      name: 'Coin98',
      image: 'https://coin98.s3.amazonaws.com/uwqLJuMfd8fsehgM',
      address: '0xaEC945e04baF28b135Fa7c640f624f8D90F1C3a6',
      isVaultCoin: true,
      isEarnCoin: true
    },
    pairs: [
      { address: '', symbol: 'BNB' },
      { address: '0x1AF3F329e8BE154074D8769D1FFa4eE058B1DBc3', symbol: 'DAI' },
      { address: '0xe9e7CEA3DedcA5984780Bafc599bD69ADd087D56', symbol: 'BUSD' },
      { address: '0x55d398326f99059fF775485246999027B3197955', symbol: 'USDT' }
    ]
  },

  [CHAIN_TYPE.binanceSmart]: {
    chain: CHAIN_TYPE.binanceSmart,
    from: {
      cgkId: 'binancecoin',
      chainId: 56,
      chain: CHAIN_TYPE.binanceSmart,
      name: 'BNB Token',
      symbol: 'BNB',
      decimals: 18,
      image: getChainImage(CHAIN_TYPE.binanceSmart),
      wrapped: '0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c'
    },
    to: {
      chainId: 56,
      chain: CHAIN_TYPE.binanceSmart,
      cgkId: 'coin98',
      symbol: 'C98',
      decimals: 18,
      name: 'Coin98',
      image: 'https://coin98.s3.amazonaws.com/uwqLJuMfd8fsehgM',
      address: '0xaEC945e04baF28b135Fa7c640f624f8D90F1C3a6',
      isVaultCoin: true,
      isEarnCoin: true
    },
    pairs: [
      { address: '', symbol: 'BNB' },
      { address: '0x1AF3F329e8BE154074D8769D1FFa4eE058B1DBc3', symbol: 'DAI' },
      { address: '0xe9e7CEA3DedcA5984780Bafc599bD69ADd087D56', symbol: 'BUSD' },
      { address: '0x55d398326f99059fF775485246999027B3197955', symbol: 'USDT' }
    ]
  },

  [CHAIN_TYPE.avax]: {
    chain: CHAIN_TYPE.avax,
    from: {
      cgkId: 'avalanche-2',
      chainId: 43114,
      chain: CHAIN_TYPE.avax,
      name: 'Avax',
      symbol: 'AVAX',
      decimals: 18,
      image: getChainImage(CHAIN_TYPE.avax),
      wrapped: '0xB31f66AA3C1e785363F0875A1B74E27b85FD66c7'
    },
    to: {
      cgkId: 'usd-coin',
      chainId: 43114,
      chain: CHAIN_TYPE.avax,
      name: 'USD Coin',
      symbol: 'USDC.e',
      address: '0xA7D7079b0FEaD91F3e65f86E8915Cb59c1a4C664',
      decimals: 6,
      image: 'https://coin98.s3.amazonaws.com/2wqqLHJ7geU17eZV'
    },
    pairs: [
      { symbol: 'AVAX', address: '' },
      { address: '0xc7198437980c041c805A1EDcbA50c1Ce5db95118', symbol: 'USDT.e' },
      { address: '0xd586E7F844cEa2F87f50152665BCbc2C279D8d70', symbol: 'DAI.e' }
    ]
  },
  [CHAIN_TYPE.matic]: {
    chain: CHAIN_TYPE.matic,
    from: {
      cgkId: 'matic-network',
      chain: CHAIN_TYPE.matic,
      chainId: 137,
      name: 'Matic',
      symbol: 'POL',
      decimals: 18,
      image: getChainImage(CHAIN_TYPE.matic),
      wrapped: '0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270'
    },
    to: {
      cgkId: 'tether',
      name: 'TetherUSD',
      symbol: 'USDT',
      address: '0xc2132D05D31c914a87C6611C10748AEb04B58e8F',
      chainId: 137,
      chain: CHAIN_TYPE.matic,
      decimals: 6,
      image: 'https://coin98.s3.amazonaws.com/n6mEe5uR8IQkt3tb'
    },
    pairs: [
      { symbol: 'MATIC', address: '' },
      { address: '0xc2132D05D31c914a87C6611C10748AEb04B58e8F', symbol: 'USDT' },
      { address: '0x831753DD7087CaC61aB5644b308642cc1c33Dc13', symbol: 'QUICK' }
    ]
  },
  [CHAIN_TYPE.fantom]: {
    chain: CHAIN_TYPE.fantom,
    from: {
      cgkId: 'fantom',
      chainId: 250,
      chain: CHAIN_TYPE.fantom,
      name: 'Fantom',
      symbol: 'FTM',
      decimals: 18,
      image: getChainImage(CHAIN_TYPE.fantom),
      wrapped: '0x21be370D5312f44cB42ce377BC9b8a0cEF1A4C83'
    },
    to: {
      cgkId: 'usd-coin',
      name: 'USD Coin',
      symbol: 'USDC',
      address: '0x28a92dde19D9989F39A49905d7C9C2FAc7799bDf',
      image: 'https://coin98.s3.amazonaws.com/2wqqLHJ7geU17eZV',
      chainId: 250,
      chain: CHAIN_TYPE.fantom,
      decimals: 6
    },
    pairs: [
      { symbol: 'FTM', address: '' },
      { address: '0x28a92dde19D9989F39A49905d7C9C2FAc7799bDf', symbol: 'USDC' },
      { address: '0x049d68029688eAbF473097a2fC38ef61633A3C7A', symbol: 'USDT' }
    ]
  },

  [CHAIN_TYPE.heco]: {
    chain: CHAIN_TYPE.heco,
    from: {
      cgkId: 'huobi-token',
      chainId: 128,
      chain: CHAIN_TYPE.heco,
      name: 'HT Token',
      symbol: 'HT',
      decimals: 18,
      image: getChainImage(CHAIN_TYPE.heco),
      wrapped: '0x5545153CCFcA01fbd7Dd11C0b23ba694D9509A6F'
    },
    to: {
      name: 'Heco-Peg HUSD Token',
      symbol: 'HUSD',
      address: '0x0298c2b32eaE4da002a15f36fdf7615BEa3DA047',
      chainId: 128,
      chain: CHAIN_TYPE.heco,
      decimals: 8,
      image: 'https://coin98.s3.amazonaws.com/ZpCQe9gYijUfxbnu',
      cgkId: 'husd'
    },
    pairs: [
      { symbol: 'HT', address: '' },
      { address: '0x0298c2b32eaE4da002a15f36fdf7615BEa3DA047', symbol: 'HUSD' },
      { address: '0xa71EdC38d189767582C38A3145b5873052c3e47a', symbol: 'USDT' }
    ]
  },
  [CHAIN_TYPE.arbitrum]: {
    chain: CHAIN_TYPE.arbitrum,
    from: {
      cgkId: 'ethereum',
      chainId: 42161,
      chain: CHAIN_TYPE.arbitrum,
      name: 'Ether',
      symbol: 'ETH',
      decimals: 18,
      image: getChainImage(CHAIN_TYPE.ether),
      wrapped: '0x82aF49447D8a07e3bd95BD0d56f35241523fBab1'
    },
    to: {
      cgkId: 'tether',
      name: 'TetherUSD',
      symbol: 'USDT',
      address: '0xFd086bC7CD5C481DCC9C85ebE478A1C0b69FCbb9',
      chainId: 42161,
      chain: CHAIN_TYPE.arbitrum,
      decimals: 6,
      image: 'https://coin98.s3.amazonaws.com/n6mEe5uR8IQkt3tb'
    },
    pairs: [
      { address: '', symbol: 'ETH' },
      { address: '0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1', symbol: 'DAI' },
      { address: '0xaf88d065e77c8cC2239327C5EDb3A432268e5831', symbol: 'USDC' },
      { address: '0xFd086bC7CD5C481DCC9C85ebE478A1C0b69FCbb9', symbol: 'USDT' }
    ]
  },
  [CHAIN_TYPE.optimism]: {
    chain: CHAIN_TYPE.optimism,
    from: {
      cgkId: 'ethereum',
      chainId: 10,
      chain: CHAIN_TYPE.optimism,
      name: 'Ether',
      symbol: 'ETH',
      decimals: 18,
      image: getChainImage(CHAIN_TYPE.ether),
      wrapped: '0x4200000000000000000000000000000000000006'
    },
    to: {
      cgkId: 'tether',
      name: 'TetherUSD',
      symbol: 'USDT',
      address: '0x94b008aA00579c1307B0EF2c499aD98a8ce58e58',
      chainId: 10,
      chain: CHAIN_TYPE.optimism,
      decimals: 6,
      image: 'https://coin98.s3.amazonaws.com/n6mEe5uR8IQkt3tb'
    },
    pairs: [
      { address: '', symbol: 'ETH' },
      { address: '0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1', symbol: 'DAI' },
      { address: '0x0b2C639c533813f4Aa9D7837CAf62653d097Ff85', symbol: 'USDC' },
      { address: '0x94b008aA00579c1307B0EF2c499aD98a8ce58e58', symbol: 'USDT' }
    ]
  },
  [CHAIN_TYPE.solana]: {
    chain: CHAIN_TYPE.solana,
    from: {
      cgkId: 'solana',
      name: 'Solana',
      chainId: 'solana',
      chain: CHAIN_TYPE.solana,
      symbol: 'SOL',
      decimals: 9,
      image: getChainImage(CHAIN_TYPE.solana),
      wrapped: 'So11111111111111111111111111111111111111112'
    },
    to: {
      cgkId: 'coin98',
      name: 'Coin98',
      chainId: 'solana',
      chain: CHAIN_TYPE.solana,
      symbol: 'C98',
      address: 'C98A4nkJXhpVZNAZdHUA95RpTF3T4whtQubL3YobiUX9',
      decimals: 6,
      image: 'https://coin98.s3.amazonaws.com/TVF7VPTqpm6bmDqg'
    }
  },
  [CHAIN_TYPE.ton]: {
    chain: CHAIN_TYPE.ton,
    from: {
      address: '',
      cgkId: 'the-open-network',
      chainId: '-239',
      chain: CHAIN_TYPE.ton,
      name: 'Ton',
      symbol: 'TON',
      decimals: 9,
      image: getChainImage(CHAIN_TYPE.ton),
      wrapped: 'Ton11111111'
    },
    to: {
      cgkId: 'tether',
      name: 'Tether USD',
      symbol: 'USDT',
      address: 'EQCxE6mUtQJKFnGfaROTKOt1lZbDiiX1kCixRv7Nw2Id_sDs',
      chainId: '-239',
      chain: CHAIN_TYPE.ton,
      decimals: 6,
      image: 'https://tether.to/images/logoCircle.png'
    }

  },
  [CHAIN_TYPE.tomo]: {
    chain: CHAIN_TYPE.tomo,
    from: {
      cgkId: 'tomochain',
      chainId: 88,
      chain: CHAIN_TYPE.tomo,
      name: 'Viction',
      symbol: 'VIC',
      decimals: 18,
      image: getChainImage(CHAIN_TYPE.tomo),
      wrapped: '0xC054751BdBD24Ae713BA3Dc9Bd9434aBe2abc1ce'
    },
    to: {
      cgkId: 'coin98',
      name: 'Coin98',
      symbol: 'C98',
      address: '0x0Fd0288AAAE91eaF935e2eC14b23486f86516c8C',
      chainId: 88,
      decimals: 18,
      chain: CHAIN_TYPE.tomo,
      image: 'https://coin98.s3.amazonaws.com/TVF7VPTqpm6bmDqg'
    }
  },
  [CHAIN_TYPE.base]: {
    chain: CHAIN_TYPE.base,
    from: {
      cgkId: 'ethereum',
      chainId: 8453,
      name: 'Ether',
      symbol: 'ETH',
      decimals: 18,
      chain: CHAIN_TYPE.base,
      image: getChainImage(CHAIN_TYPE.ether),
      wrapped: '0x4200000000000000000000000000000000000006'
    },
    to: {
      cgkId: 'tether',
      name: 'USD Coin',
      symbol: 'USDC',
      address: '0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913',
      chainId: 8453,
      decimals: 6,
      chain: CHAIN_TYPE.base,
      image: 'https://coin98.s3.amazonaws.com/2wqqLHJ7geU17eZV'
    }
  }

}

export const DEFAULT_EXCHANGE = CHAIN_TYPE.ether

export const FIXED_STABLE_BY_CHAIN: FixedStableList = {
  ether: '0xdAC17F958D2ee523a2206206994597C13D831ec7',
  binanceSmart: '0x1AF3F329e8BE154074D8769D1FFa4eE058B1DBc3',
  fantom: '0x04068DA6C83AFCFA0e13ba15A6696662335D5B75',
  matic: '0xc2132D05D31c914a87C6611C10748AEb04B58e8F',
  heco: '0x0298c2b32eaE4da002a15f36fdf7615BEa3DA047',
  avax: '0xc7198437980c041c805A1EDcbA50c1Ce5db95118',
  arbitrum: '0xFd086bC7CD5C481DCC9C85ebE478A1C0b69FCbb9',
  optimism: '0x94b008aA00579c1307B0EF2c499aD98a8ce58e58',
  solana: '',
  allChain: '',
  ton: '',
  tomo: '',
  base: ''
}

export const EXCHANGE_LIST = Object.keys(EXCHANGE_TYPE)
